packages/types/src/auth.ts

```diff
@@ -25,3 +25,37 @@ export interface AuthScheme {
   signingScope?: never;
   properties: Record<string, unknown>;
 }
+
+// As described in the Smithy documentation:
+// https://github.com/awslabs/smithy/blob/main/smithy-model/src/main/resources/software/amazon/smithy/model/loader/prelude.smithy
```

Comment:
Can you use the same interface as in Smithy?

```ts
export interface HttpApiKeyAuth {
  /**
   * Defines the name of the HTTP header or query string parameter
   * that contains the API key.
   */
  name: string;

  /**
   * Defines the location of where the key is serialized.
   */
  in: HttpApiKeyLocation;

  /**
   * Defines the security scheme to use on the `Authorization` header value.
   * This can only be set if the "in" property is set to {@link HttpApiKeyLocation.HEADER}.
   */
  scheme?: string,
}

export enum HttpApiKeyLocation {
  HEADER = "header",
  QUERY = "query"
}
```

Reply:
I have created the enum for the location, but for the rest we are using those interfaces for both [API Key](https://github.com/aws/aws-sdk-js-v3/blob/4bc9f53335423221f2f2183a0dab462a0564bf4f/packages/middleware-api-key/src/middleware.ts#L31) and [Bearer token](https://github.com/aws/aws-sdk-js-v3/blob/4bc9f53335423221f2f2183a0dab462a0564bf4f/packages/middleware-token/src/tokenMiddleware.ts#L33-L37) usage.